tests/integration/Core/Framework/DataAbstractionLayer/Dbal/EntityReaderTest.php

```diff
@@ -0,0 +1,304 @@
+<?php
+/*
+ * Copyright (c) Pickware GmbH. All rights reserved.
+ * This file is part of software that is released under a proprietary license.
+ * You must not copy, modify, distribute, make publicly available, or execute
+ * its contents or parts thereof without express permission by the copyright
+ * holder, unless otherwise permitted by law.
+ */
+
+declare(strict_types=1);
+
+namespace Shopware\Tests\Integration\Core\Framework\DataAbstractionLayer\Dbal;
+
+use Doctrine\DBAL\Connection;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Content\Product\ProductCollection;
+use Shopware\Core\Content\Product\ProductDefinition;
+use Shopware\Core\Content\Test\Product\ProductBuilder;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Api\Context\SystemSource;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\CriteriaFieldsResolver;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\CriteriaQueryBuilder;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\EntityDefinitionQueryHelper;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\EntityHydrator;
+use Shopware\Core\Framework\DataAbstractionLayer\Dbal\EntityReader;
+use Shopware\Core\Framework\DataAbstractionLayer\Entity;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Parser\SqlQueryParser;
+use Shopware\Core\Framework\Test\IdsCollection;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+
+/**
+ * @internal
+ */
+#[CoversClass(EntityReader::class)]
+class EntityReaderTest extends TestCase
+{
+    use IntegrationTestBehaviour;
+
+    private EntityReader $entityReader;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->entityReader = new EntityReader(
+            $this->getContainer()->get(Connection::class),
+            $this->getContainer()->get(EntityHydrator::class),
+            $this->getContainer()->get(EntityDefinitionQueryHelper::class),
+            $this->getContainer()->get(SqlQueryParser::class),
+            $this->getContainer()->get(CriteriaQueryBuilder::class),
+            $this->getContainer()->get('logger'),
+            $this->getContainer()->get(CriteriaFieldsResolver::class)
+        );
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        $this->getContainer()->get(Connection::class)->executeQuery('SET FOREIGN_KEY_CHECKS=1;');
+    }
+
+    public function testReadLoadsTranslationsAssociations(): void
+    {
+        $productId = $this->createProduct(
+            deDeTranslation: 'Deutscher Name',
+            defaultTranslation: null,
+        );
+
+        $criteria = new Criteria();
+        $criteria->addAssociations(['translations']);
+
+        /** @var ProductCollection $products */
```

Comment:
please do not use var annotations in tests. all types could be verified with `assert*` methods